riptide-failsafe/src/main/java/org/zalando/riptide/soap/PreserveContextClassLoaderTaskDecorator.java

```diff
@@ -0,0 +1,29 @@
+package org.zalando.riptide.soap;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import net.jodah.failsafe.function.ContextualSupplier;
+import org.apiguardian.api.API;
+import org.zalando.riptide.failsafe.TaskDecorator;
+
+/**
+ * @see <a href="https://github.com/zalando/riptide/issues/953">JAXB + ForkJoinPool</a>
+ */
+@API(status = EXPERIMENTAL)
+public class PreserveContextClassLoaderTaskDecorator implements TaskDecorator {
+
+  private final ClassLoader loader = Thread.currentThread().getContextClassLoader();
```

Comment:
This might not be the class loader that we're looking for. We create task decorators in the main thread, when we wire up the Riptide client. I think it would be cleaner if the current class loader is read within `decorate` but before the closure that is returned.

Reply:
That what I thought as well.. I will think about reworking